icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -797,10 +798,9 @@ MeasureUnit MeasureUnit::simplify(UErrorCode &status) const {
 
     MeasureUnitImpl resultImpl;
 
-    int32_t outCount;
-    auto singleUnits = this->splitToSingleUnits(outCount, status);
-    for (int i = 0; i < outCount; ++i) {
-        findAndMerge(resultImpl, SingleUnitImpl::forMeasureUnit(singleUnits[i], status), status);
+    const auto &units = (*MeasureUnitImpl::get(*this)).units;
```

Comment:
This will segfault if the target unit is a built-in unit.  Use this instead:

```suggestion
    MeasureUnitImpl temp;
    const auto &units = MeasureUnitImpl::forMeasureUnit(other, temp, status).units;
```

Reply:
Done, thanks for the info.